cmd/cluster/create.go

```
@@ -78,6 +81,68 @@ func createCluster(config *types.ClusterConfig) error {
 		return fmt.Errorf("multipass is not installed or not in PATH")
 	}
 
+	if !skipValidation {
+		requirements, err := validator.CalculateResourceRequirements(
+			config.MasterCPUs, config.MasterMemory, config.MasterDisk,
+			config.WorkerCPUs, config.WorkerMemory, config.WorkerDisk,
+			config.Size-1,
+		)
+		if err != nil {
+			return fmt.Errorf("failed to calculate resource requirements: %w", err)
+		}
+
+		status, err := validator.ValidateResources(requirements)
+		if err != nil {
+			return fmt.Errorf("failed to validate host resources: %w", err)
+		}
+
+		portStatus, err := validator.ValidatePorts()
```

Could you explain what is the problem on Ports, is not  `multipass` ensure dedicated IP address for each node ?

---

As you mention in issue feature file, you want to validate if port is available? , if not -> `fail fast`. 
That's  what i tried  to achieve, I don't know where `multipass` check for port in code for sure.

---

Ah, please keep in mind that when I outline a new issue or feature, it's not necessarily the finalized design. Some assumptions may be inaccurate or incomplete, so it's important that you validate and critically evaluate the requirements before proceeding